Update ANGLE to 2022-09-12 (26cb336a0dfcd4cd657a2b12ccf020779682e718)

kkinnunen-apple opened the pull request:
#### 7ab8cd2a16e426ab14810f92faf97f95c2c1053c
<pre>
Update ANGLE to 2022-09-12 (26cb336a0dfcd4cd657a2b12ccf020779682e718)
<a href="https://bugs.webkit.org/show_bug.cgi?id=245072">https://bugs.webkit.org/show_bug.cgi?id=245072</a>
rdar://problem/99819001&gt;

Reviewed by Kenneth Russell.

Contains upstream commits:
git log --oneline 92e4cd75785d257a5020b197ca382b06c32e9722..26cb336a0dfcd4cd657a2b12ccf020779682e718 --pretty=%h %s
26cb336a0 Roll vulkan-deps from b772fe545415 to 3811e73c4b6f (4 revisions)
a25f92882 Roll SwiftShader from 27383fa99bbd to 4258f175b8d9 (3 revisions)
09446a6bf Metal: Upload the last texture row separately on AMD.
6c469cdfb Properly Create C++ string objects
8f0e104dd Roll SwiftShader from 4e4d2b45391f to 27383fa99bbd (1 revision)
69cee5c22 Roll vulkan-deps from 6feded35cfee to b772fe545415 (6 revisions)
585d2a9e5 Capture/Replay: Capture the attr locations as set by the program
8fdfd8be4 Update Dev docs to mention Chromium
6937ea98c Use base name of the generator script in various generated files
191175676 Avoid general python exceptions in retrace script
daa134d90 GL: Ignore qualcomm errors about timestamps
61e905afe Move out of bounds buffer access tests under correct extension
526e8e647 Fix UNSYNCHRONIZED glMapBufferRange test
6387f4118 Roll VK-GL-CTS from f08773d25d83 to 2d84fcfd2251 (4 revisions)
313956f2b Vulkan: Don&apos;t store EGLSyncVk::mAttribs as a const ref.
5fae671c1 Vulkan: Enable VK_EXT_swapchain_colorspace when supported
218a0c241 Roll SwiftShader from 7c6c7b79af8c to 4e4d2b45391f (6 revisions)
88287eac0 Roll vulkan-deps from c8b855ee4e1d to 6feded35cfee (4 revisions)
11fd8346d Metal: Make flush NoWait
f99e255b4 GL: Unbind buffers after mapping them.
b9df95ffd Add option to enable System EGL on all test configs
490ec5ba4 Replace underscores with dashes in SYNC-HAZARD messages
79aa846e3 Reland &quot;Not recreate Framebuffer for eglMakeCurrent() call&quot;
2deae29ea Change xvfb screen to 3120x3120.
f5b3ecd46 fix perf test result path
9f7b6babe Add minimal setup for Go codegen in Android.bp.
76a74d85a Roll SwiftShader from be37d7fbc94f to 7c6c7b79af8c (1 revision)
eff98f6c2 Roll vulkan-deps from e3fa08b13e32 to c8b855ee4e1d (8 revisions)
02e8497f1 Revert &quot;Not recreate Framebuffer for eglMakeCurrent() call&quot;
f1215790f Roll Chromium from 1c4ee1412503 to e7211d52993a (477 revisions)
fa6b6d174 Fix the SYNC-HAZARD error message format
91deedba5 Vulkan: Use VK_EXT_pipeline_robustness
1d04539fc Fix xfb tests rendering points
9c17232a8 Vulkan: Make robustness affect all of share group
2ede53ec7 Replace underscores with dashes in SYNC-HAZARD messages
662226a32 Metal: Preemptively Start Provoking Vertex CmdBuffer on AMD
ff46a03a8 Vulkan: Use VK_EXT_multisampled_render_to_single_sampled
9aeb6311a Revert &quot;Extra buffer logging/checking on Android Cuttlefish&quot;
2c4f71871 Metal: Stop pre-creating the Provoking Vertex CmdBuffer
f859f97f7 Use the same build_info in all functions and cache nodes.
bf9c81526 Not recreate Framebuffer for eglMakeCurrent() call
1cac338a3 FrameCapture: Fix GLES1 vertex array state
8454b625e Roll vulkan-deps from c4e128e05c38 to e3fa08b13e32 (12 revisions)
c2ad74e4e Roll Chromium from c709ec453fcb to 1c4ee1412503 (1424 revisions)
1b39c610c Roll SwiftShader from f0db6df92f88 to be37d7fbc94f (6 revisions)
ba721efd6 Vulkan: FIx Android AHB format mapping
3960e63b1 Infra: Enable angle_deqp_gl46_tests on SwiftShader
80f6a63af Tests external image uploads with colorspace
2e142b7f0 Roll vulkan-deps from 40446ec62941 to c4e128e05c38 (14 revisions)
ba65030c2 Fix adding imageless FBO to enabled extensions
ba87b1950 OpenGL: Pass EGL_GL_COLORSPACE_KHR to OGL backend
e8b6cd217 Skip VulkanPerformanceCounterTest.EndXfbAfterRenderPassClosed
61d5340d2 Skip EmulateCopyTexImage2DFromRenderbuffers tests on Win/Intel
d9f5b2f89 Vulkan: Dont skip texture upload if only color space differs
6ec89510f Frame Capture: Remove MEC active variable.
ae7481a60 Move entry point classifier helpers into angle_common.
38a38b8de Revert &quot;EndXfbAfterRenderPassClosed expectation (0,0) -&gt; (w/2,h/2)&quot;
baf5356c6 Map RGBX AHB back to GL_RGB8 for OpenGL
2dc1c609d EndXfbAfterRenderPassClosed expectation (0,0) -&gt; (w/2,h/2)
f2f17bad8 Vulkan: Fix Framebuffer Fetch Test Failure
d0fad24ce Add noncoherent PLS tests
83293767b Roll SwiftShader from ecba967ea71f to f0db6df92f88 (5 revisions)
cec499f28 Roll vulkan-deps from da42de1e5ad7 to 40446ec62941 (12 revisions)
4d3c50f3c Roll Chromium from 6511404d8c85 to c709ec453fcb (552 revisions)
c6ad305cc Vulkan: No depth load/store if depthFunc==ALWAYS/NEVER &amp;&amp; mask==FALSE
2f9d0810e Revert &quot;Sort items to avoid python2/3 diffs.&quot;
b43a00456 Sort items to avoid python2/3 diffs.
f602278c2 Vulkan: Parse human readable ARM driver versions.
f2c01af23 Roll vulkan-deps from db1584efaa9b to da42de1e5ad7 (1 revision)
a28b81354 Roll SwiftShader from 476165cc7c0c to ecba967ea71f (4 revisions)
080b67ed0 Roll Chromium from 25d1a1a7a886 to 6511404d8c85 (577 revisions)
e9cb83dd9 Vulkan: disable EGL_EXT_image_gl_colorspace
7428369ad Vulkan: Use macros for load/store Op check
c8b0caf1f Vulkan: Don&apos;t try to guess format mapping
00411e2c1 Support provoking vertex state query
c10326f14 Make shader dump code compile after ShCompileOptions API change
2c3737719 Add include atomic, missing in Skia builds.
7788fb6a8 Roll vulkan-deps from 7d94bcff7a00 to db1584efaa9b (3 revisions)
6d43eb627 Roll Chromium from 75cc1cb02dd7 to 25d1a1a7a886 (533 revisions)
70c434b20 Check max shader storage blocks support in test
ac19624e9 Vulkan: Handle suboptimal like out-of-date
64f41972a Use canonical gl.xml and update enum to string function.
320729055 d3d11: Support rendering to GL_TEXTURE_2D target for YUV images
a18a04147 Roll VK-GL-CTS from 435ebaff9214 to f08773d25d83 (15 revisions)
f18b6335e Replace std::this_thread::get_id() use with a unique thread id.
9fb05a883 Vulkan: SPIR-V Gen: Remove suppression of passing tests
b9df3551d update CopyTextureCHROMIUM to match Chromium
4b33633a7 Capture/Replay: Delete binary data in FinishReplay via callback
cd0f642e4 Fix loading end2end test expectations on iOS
ba4b69130 Fix data race in BlobCache
5ef3960bc Use correct seed type for XXH32 fuzzer
c63f9bcc5 Roll vulkan-deps from eb46f95f20dc to 7d94bcff7a00 (8 revisions)
4d182868a Roll SwiftShader from 6517262b2386 to 476165cc7c0c (6 revisions)
0533280f7 Roll Chromium from 21906127bee6 to 75cc1cb02dd7 (625 revisions)
bec40d768 Vulkan: Set SPIR-V Generator version to 1
ac2aeccde Check GL_MAX_VERTEX_SHADER_STORAGE_BLOCKS in test
cb618b3d4 Vulkan: Remove useRelaxedPrecision
2d48a3707 Vulkan: Support passthrough of colorspace attributes
98d5db706 Add capture functions for desktop GL entry points
6ba8582b9 Add a fix on all backends for images as function arguments
40782cf56 Roll Chromium from 32095292c129 to 21906127bee6 (499 revisions)
efea3934b Roll vulkan-deps from e623f608cbf3 to eb46f95f20dc (4 revisions)
02eb5f490 validation: Annotate kEnumNotSupported errors with values.
320bfde0f Hide emulated ETC2/EAC formats from WebGL contexts
9d41585e1 Make PLS coherent on D3D 11.3
cedc23d81 validation: Annotate kInvalidInternalFormat errors with values.
51de38374 Enable PLS on D3D
f6842cc2c Revert override for desktop GL extensions in gl.xml.
5bcfad04d Vulkan: Add a handful of features
68b6206ef FrameCapture: Fix capturing glBindRenderbuffer on GLES1.
80a9ee718 Fix translator fuzzer
a35728bda Skip several Multithreading Tests on SwiftShader.
5d62c05e2 Use canonical Khronos registry XML files for auto-gen.
b688c9098 Capture/Replay: save and restore UNPACK_ALIGNMENT in MEC
a69e53c56 Vulkan: Move vertex stride assert
15b1678f2 FrameCapture: Add VertexArrayCaptureFuncs.
ab9fddb78 Roll SwiftShader from 67f88061d2d4 to 6517262b2386 (3 revisions)
ccbc0289e Roll vulkan-deps from 188a44aaa198 to e623f608cbf3 (2 revisions)
f556b9188 Roll Chromium from 5377f9824cad to 32095292c129 (473 revisions)
c394adfc2 ANGLE: reorder program and shader creation
efa9a4d91 Rename README.angle to README.chromium
b9bc09cbc Vulkan: Always enable mVirtualBlockMutex
d89c027e7 Add unimplemented GLX entry points to desktop GL
91850a24e Add flaky reties to Mac GL dEQP-GLES 2 &amp; 3 tests.
6e784d9c0 Roll Chromium from 02dbf021e1a1 to 5377f9824cad (537 revisions)
28600f606 Test: Add Antutu Refinery trace
3e9d8f435 Fix build with -Wextra-semi
1393ba6d3 Roll vulkan-deps from 1b46e43d5ccc to 188a44aaa198 (8 revisions)
f1f7f3a2c Roll SwiftShader from c21aa26e0256 to 67f88061d2d4 (1 revision)
5317b778e Enable desktop GLSL for desktop GL frontend
f7274de21 Skip EGLMultiContextTest.ReuseUnterminatedDisplay on Pixel 4 GLES
94a23a4ed FrameCapture: Fix separable shader uniform locs
429a7cabd FrameCapture: Fix cube face as framebuffer
aa2a558ec Vulkan: Add support for setting timestamp surface attribute
f10625d3c Enable Robustness Extension on ARM
13456bfc2 Propagate device type in another eglGetPlatformDisplayEXT call.
f0b4e8b2e Fix loading ANGLE libraries on iOS
41c01134a Extend the image test to verify RGBX AHB and GL_RGB upload
df9468565 Map RGBX AHB to GL_RGBX8_ANGLE and fix the load function
eab2d546e Roll vulkan-deps from 186a4d4b9dea to 1b46e43d5ccc (2 revisions)
7131475ee Roll Chromium from e1b4daf2a3e7 to 02dbf021e1a1 (568 revisions)
6fde3568a Spec out the PLS client API
b32091b42 Vulkan: Bug fix related to pre-rotation code
952d6e8f8 Vulkan: Workaround ARM driver bug with dynamic vertex strides
bd1484f3a Temporarily disable shader caching feature
86a35a4b5 Roll vulkan-deps from d53c356ad7d9 to 186a4d4b9dea (48 revisions)
e22eb503c Metal: Report Metal being available on iOS simulator
a8ffbef8c Roll VK-GL-CTS from 5a93cd60f33b to 435ebaff9214 (5 revisions)
caf2b4279 Vulkan: Add feature to use PASS_THROUGH for EGL_NONE color spaces
f30cfd180 Perf and gold tests detect test SKIPs from json results.
4b4c90efb Roll SwiftShader from 1c3dfde53353 to c21aa26e0256 (3 revisions)
676bc8ed4 Roll Chromium from 336269709208 to e1b4daf2a3e7 (602 revisions)
31ac3fefd Lift suppression of passing test
b9c3c5d67 Metal: Fix MTLArgumentEncoder leak in InitArgumentBufferEncoder
9a2582811 Fix submit-count perf counter test on ARM
4330a827b Reland &quot;Vulkan: Use VK_EXT_primitive_topology_list_restart&quot;
b8b44fc85 Roll SwiftShader from ab4bf6ad00d4 to 1c3dfde53353 (1 revision)
8ef496b15 Roll Chromium from 7f3befa9aba1 to 336269709208 (495 revisions)
6d153b7d7 Implement BaseInstanceEXT Capture Entry Points
29d18b303 Revert &quot;Vulkan: Use VK_EXT_primitive_topology_list_restart&quot;
eb8488e1b Skip end2end failures on Pixel 6 Android 13 TP1A.220624.021
c6c7ff3c4 Roll SwiftShader from 5ac9bd16c803 to ab4bf6ad00d4 (1 revision)
f5698a8f6 Roll Chromium from 754ede9fc072 to 7f3befa9aba1 (1095 revisions)
955adb775 Cache compiled shader
2965f81b8 Remove suppression of previously crashed deqp test
266d0a831 Skip glObjectLabel in replay capture
eb535debc Don&apos;t set the framebuffer read buffer dirty bit on ES2.
7a87e90d0 An inactive but alive thread shouldn&apos;t prevent cleanup
8fe5fd578 Reland Copy .vpython3 from chromium/src/+/main:.vpython3
0951686d9 Add .vpython .vpython3 to trigger all tests on changes there.
24012abcb Revert &quot;Copy .vpython3 from chromium/src/+/main:.vpython3&quot;
3b91270fb Copy .vpython3 from chromium/src/+/main:.vpython3
6a8bd8010 Add process_angle_perf_results.py to perf tests data.

Canonical link: <a href="https://commits.webkit.org/254481@main">https://commits.webkit.org/254481@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4fa07be4a7e37f6568aa2dc94e197587bdec0c43

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/89180 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/33741 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/20003 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/98500 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/154812 "Built successfully and passed tests") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/32249 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/27801 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/81543 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/92969 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/94827 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/68/builds/25616 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/76109 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/25550 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/80481 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/80455 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/68524 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/80890 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/30029 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/14519 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/74681 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/29754 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/15483 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/26322 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3138 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/33200 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/38448 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/77546 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/31900 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/34574 "Passed tests") | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/17158 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

kenrussell reviewed (approved):
Looks good to me. r+

I reviewed everything in changes.diff and src/libANGLE/renderer/metal.

As mentioned earlier - do watch out for the semantic change in the Metal backend's glFlush implementation. Hopefully this won't cause any incorrect IOSurface sychronization in WebKit.

kkinnunen-apple committed:
Update ANGLE to 2022-09-12 (26cb336a0dfcd4cd657a2b12ccf020779682e718)
https://bugs.webkit.org/show_bug.cgi?id=245072
rdar://problem/99819001>

Reviewed by Kenneth Russell.

Contains upstream commits:
git log --oneline 92e4cd75785d257a5020b197ca382b06c32e9722..26cb336a0dfcd4cd657a2b12ccf020779682e718 --pretty=%h %s
26cb336a0 Roll vulkan-deps from b772fe545415 to 3811e73c4b6f (4 revisions)
a25f92882 Roll SwiftShader from 27383fa99bbd to 4258f175b8d9 (3 revisions)
09446a6bf Metal: Upload the last texture row separately on AMD.
6c469cdfb Properly Create C++ string objects
8f0e104dd Roll SwiftShader from 4e4d2b45391f to 27383fa99bbd (1 revision)
69cee5c22 Roll vulkan-deps from 6feded35cfee to b772fe545415 (6 revisions)
585d2a9e5 Capture/Replay: Capture the attr locations as set by the program
8fdfd8be4 Update Dev docs to mention Chromium
6937ea98c Use base name of the generator script in various generated files
191175676 Avoid general python exceptions in retrace script
daa134d90 GL: Ignore qualcomm errors about timestamps
61e905afe Move out of bounds buffer access tests under correct extension
526e8e647 Fix UNSYNCHRONIZED glMapBufferRange test
6387f4118 Roll VK-GL-CTS from f08773d25d83 to 2d84fcfd2251 (4 revisions)
313956f2b Vulkan: Don't store EGLSyncVk::mAttribs as a const ref.
5fae671c1 Vulkan: Enable VK_EXT_swapchain_colorspace when supported
218a0c241 Roll SwiftShader from 7c6c7b79af8c to 4e4d2b45391f (6 revisions)
88287eac0 Roll vulkan-deps from c8b855ee4e1d to 6feded35cfee (4 revisions)
11fd8346d Metal: Make flush NoWait
f99e255b4 GL: Unbind buffers after mapping them.
b9df95ffd Add option to enable System EGL on all test configs
490ec5ba4 Replace underscores with dashes in SYNC-HAZARD messages
79aa846e3 Reland "Not recreate Framebuffer for eglMakeCurrent() call"
2deae29ea Change xvfb screen to 3120x3120.
f5b3ecd46 fix perf test result path
9f7b6babe Add minimal setup for Go codegen in Android.bp.
76a74d85a Roll SwiftShader from be37d7fbc94f to 7c6c7b79af8c (1 revision)
eff98f6c2 Roll vulkan-deps from e3fa08b13e32 to c8b855ee4e1d (8 revisions)
02e8497f1 Revert "Not recreate Framebuffer for eglMakeCurrent() call"
f1215790f Roll Chromium from 1c4ee1412503 to e7211d52993a (477 revisions)
fa6b6d174 Fix the SYNC-HAZARD error message format
91deedba5 Vulkan: Use VK_EXT_pipeline_robustness
1d04539fc Fix xfb tests rendering points
9c17232a8 Vulkan: Make robustness affect all of share group
2ede53ec7 Replace underscores with dashes in SYNC-HAZARD messages
662226a32 Metal: Preemptively Start Provoking Vertex CmdBuffer on AMD
ff46a03a8 Vulkan: Use VK_EXT_multisampled_render_to_single_sampled
9aeb6311a Revert "Extra buffer logging/checking on Android Cuttlefish"
2c4f71871 Metal: Stop pre-creating the Provoking Vertex CmdBuffer
f859f97f7 Use the same build_info in all functions and cache nodes.
bf9c81526 Not recreate Framebuffer for eglMakeCurrent() call
1cac338a3 FrameCapture: Fix GLES1 vertex array state
8454b625e Roll vulkan-deps from c4e128e05c38 to e3fa08b13e32 (12 revisions)
c2ad74e4e Roll Chromium from c709ec453fcb to 1c4ee1412503 (1424 revisions)
1b39c610c Roll SwiftShader from f0db6df92f88 to be37d7fbc94f (6 revisions)
ba721efd6 Vulkan: FIx Android AHB format mapping
3960e63b1 Infra: Enable angle_deqp_gl46_tests on SwiftShader
80f6a63af Tests external image uploads with colorspace
2e142b7f0 Roll vulkan-deps from 40446ec62941 to c4e128e05c38 (14 revisions)
ba65030c2 Fix adding imageless FBO to enabled extensions
ba87b1950 OpenGL: Pass EGL_GL_COLORSPACE_KHR to OGL backend
e8b6cd217 Skip VulkanPerformanceCounterTest.EndXfbAfterRenderPassClosed
61d5340d2 Skip EmulateCopyTexImage2DFromRenderbuffers tests on Win/Intel
d9f5b2f89 Vulkan: Dont skip texture upload if only color space differs
6ec89510f Frame Capture: Remove MEC active variable.
ae7481a60 Move entry point classifier helpers into angle_common.
38a38b8de Revert "EndXfbAfterRenderPassClosed expectation (0,0) -> (w/2,h/2)"
baf5356c6 Map RGBX AHB back to GL_RGB8 for OpenGL
2dc1c609d EndXfbAfterRenderPassClosed expectation (0,0) -> (w/2,h/2)
f2f17bad8 Vulkan: Fix Framebuffer Fetch Test Failure
d0fad24ce Add noncoherent PLS tests
83293767b Roll SwiftShader from ecba967ea71f to f0db6df92f88 (5 revisions)
cec499f28 Roll vulkan-deps from da42de1e5ad7 to 40446ec62941 (12 revisions)
4d3c50f3c Roll Chromium from 6511404d8c85 to c709ec453fcb (552 revisions)
c6ad305cc Vulkan: No depth load/store if depthFunc==ALWAYS/NEVER && mask==FALSE
2f9d0810e Revert "Sort items to avoid python2/3 diffs."
b43a00456 Sort items to avoid python2/3 diffs.
f602278c2 Vulkan: Parse human readable ARM driver versions.
f2c01af23 Roll vulkan-deps from db1584efaa9b to da42de1e5ad7 (1 revision)
a28b81354 Roll SwiftShader from 476165cc7c0c to ecba967ea71f (4 revisions)
080b67ed0 Roll Chromium from 25d1a1a7a886 to 6511404d8c85 (577 revisions)
e9cb83dd9 Vulkan: disable EGL_EXT_image_gl_colorspace
7428369ad Vulkan: Use macros for load/store Op check
c8b0caf1f Vulkan: Don't try to guess format mapping
00411e2c1 Support provoking vertex state query
c10326f14 Make shader dump code compile after ShCompileOptions API change
2c3737719 Add include atomic, missing in Skia builds.
7788fb6a8 Roll vulkan-deps from 7d94bcff7a00 to db1584efaa9b (3 revisions)
6d43eb627 Roll Chromium from 75cc1cb02dd7 to 25d1a1a7a886 (533 revisions)
70c434b20 Check max shader storage blocks support in test
ac19624e9 Vulkan: Handle suboptimal like out-of-date
64f41972a Use canonical gl.xml and update enum to string function.
320729055 d3d11: Support rendering to GL_TEXTURE_2D target for YUV images
a18a04147 Roll VK-GL-CTS from 435ebaff9214 to f08773d25d83 (15 revisions)
f18b6335e Replace std::this_thread::get_id() use with a unique thread id.
9fb05a883 Vulkan: SPIR-V Gen: Remove suppression of passing tests
b9df3551d update CopyTextureCHROMIUM to match Chromium
4b33633a7 Capture/Replay: Delete binary data in FinishReplay via callback
cd0f642e4 Fix loading end2end test expectations on iOS
ba4b69130 Fix data race in BlobCache
5ef3960bc Use correct seed type for XXH32 fuzzer
c63f9bcc5 Roll vulkan-deps from eb46f95f20dc to 7d94bcff7a00 (8 revisions)
4d182868a Roll SwiftShader from 6517262b2386 to 476165cc7c0c (6 revisions)
0533280f7 Roll Chromium from 21906127bee6 to 75cc1cb02dd7 (625 revisions)
bec40d768 Vulkan: Set SPIR-V Generator version to 1
ac2aeccde Check GL_MAX_VERTEX_SHADER_STORAGE_BLOCKS in test
cb618b3d4 Vulkan: Remove useRelaxedPrecision
2d48a3707 Vulkan: Support passthrough of colorspace attributes
98d5db706 Add capture functions for desktop GL entry points
6ba8582b9 Add a fix on all backends for images as function arguments
40782cf56 Roll Chromium from 32095292c129 to 21906127bee6 (499 revisions)
efea3934b Roll vulkan-deps from e623f608cbf3 to eb46f95f20dc (4 revisions)
02eb5f490 validation: Annotate kEnumNotSupported errors with values.
320bfde0f Hide emulated ETC2/EAC formats from WebGL contexts
9d41585e1 Make PLS coherent on D3D 11.3
cedc23d81 validation: Annotate kInvalidInternalFormat errors with values.
51de38374 Enable PLS on D3D
f6842cc2c Revert override for desktop GL extensions in gl.xml.
5bcfad04d Vulkan: Add a handful of features
68b6206ef FrameCapture: Fix capturing glBindRenderbuffer on GLES1.
80a9ee718 Fix translator fuzzer
a35728bda Skip several Multithreading Tests on SwiftShader.
5d62c05e2 Use canonical Khronos registry XML files for auto-gen.
b688c9098 Capture/Replay: save and restore UNPACK_ALIGNMENT in MEC
a69e53c56 Vulkan: Move vertex stride assert
15b1678f2 FrameCapture: Add VertexArrayCaptureFuncs.
ab9fddb78 Roll SwiftShader from 67f88061d2d4 to 6517262b2386 (3 revisions)
ccbc0289e Roll vulkan-deps from 188a44aaa198 to e623f608cbf3 (2 revisions)
f556b9188 Roll Chromium from 5377f9824cad to 32095292c129 (473 revisions)
c394adfc2 ANGLE: reorder program and shader creation
efa9a4d91 Rename README.angle to README.chromium
b9bc09cbc Vulkan: Always enable mVirtualBlockMutex
d89c027e7 Add unimplemented GLX entry points to desktop GL
91850a24e Add flaky reties to Mac GL dEQP-GLES 2 & 3 tests.
6e784d9c0 Roll Chromium from 02dbf021e1a1 to 5377f9824cad (537 revisions)
28600f606 Test: Add Antutu Refinery trace
3e9d8f435 Fix build with -Wextra-semi
1393ba6d3 Roll vulkan-deps from 1b46e43d5ccc to 188a44aaa198 (8 revisions)
f1f7f3a2c Roll SwiftShader from c21aa26e0256 to 67f88061d2d4 (1 revision)
5317b778e Enable desktop GLSL for desktop GL frontend
f7274de21 Skip EGLMultiContextTest.ReuseUnterminatedDisplay on Pixel 4 GLES
94a23a4ed FrameCapture: Fix separable shader uniform locs
429a7cabd FrameCapture: Fix cube face as framebuffer
aa2a558ec Vulkan: Add support for setting timestamp surface attribute
f10625d3c Enable Robustness Extension on ARM
13456bfc2 Propagate device type in another eglGetPlatformDisplayEXT call.
f0b4e8b2e Fix loading ANGLE libraries on iOS
41c01134a Extend the image test to verify RGBX AHB and GL_RGB upload
df9468565 Map RGBX AHB to GL_RGBX8_ANGLE and fix the load function
eab2d546e Roll vulkan-deps from 186a4d4b9dea to 1b46e43d5ccc (2 revisions)
7131475ee Roll Chromium from e1b4daf2a3e7 to 02dbf021e1a1 (568 revisions)
6fde3568a Spec out the PLS client API
b32091b42 Vulkan: Bug fix related to pre-rotation code
952d6e8f8 Vulkan: Workaround ARM driver bug with dynamic vertex strides
bd1484f3a Temporarily disable shader caching feature
86a35a4b5 Roll vulkan-deps from d53c356ad7d9 to 186a4d4b9dea (48 revisions)
e22eb503c Metal: Report Metal being available on iOS simulator
a8ffbef8c Roll VK-GL-CTS from 5a93cd60f33b to 435ebaff9214 (5 revisions)
caf2b4279 Vulkan: Add feature to use PASS_THROUGH for EGL_NONE color spaces
f30cfd180 Perf and gold tests detect test SKIPs from json results.
4b4c90efb Roll SwiftShader from 1c3dfde53353 to c21aa26e0256 (3 revisions)
676bc8ed4 Roll Chromium from 336269709208 to e1b4daf2a3e7 (602 revisions)
31ac3fefd Lift suppression of passing test
b9c3c5d67 Metal: Fix MTLArgumentEncoder leak in InitArgumentBufferEncoder
9a2582811 Fix submit-count perf counter test on ARM
4330a827b Reland "Vulkan: Use VK_EXT_primitive_topology_list_restart"
b8b44fc85 Roll SwiftShader from ab4bf6ad00d4 to 1c3dfde53353 (1 revision)
8ef496b15 Roll Chromium from 7f3befa9aba1 to 336269709208 (495 revisions)
6d153b7d7 Implement BaseInstanceEXT Capture Entry Points
29d18b303 Revert "Vulkan: Use VK_EXT_primitive_topology_list_restart"
eb8488e1b Skip end2end failures on Pixel 6 Android 13 TP1A.220624.021
c6c7ff3c4 Roll SwiftShader from 5ac9bd16c803 to ab4bf6ad00d4 (1 revision)
f5698a8f6 Roll Chromium from 754ede9fc072 to 7f3befa9aba1 (1095 revisions)
955adb775 Cache compiled shader
2965f81b8 Remove suppression of previously crashed deqp test
266d0a831 Skip glObjectLabel in replay capture
eb535debc Don't set the framebuffer read buffer dirty bit on ES2.
7a87e90d0 An inactive but alive thread shouldn't prevent cleanup
8fe5fd578 Reland Copy .vpython3 from chromium/src/+/main:.vpython3
0951686d9 Add .vpython .vpython3 to trigger all tests on changes there.
24012abcb Revert "Copy .vpython3 from chromium/src/+/main:.vpython3"
3b91270fb Copy .vpython3 from chromium/src/+/main:.vpython3
6a8bd8010 Add process_angle_perf_results.py to perf tests data.

Canonical link: https://commits.webkit.org/254481@main
